src/functions/DidSiopMetadata.ts

```diff
@@ -0,0 +1,46 @@
+import { SIOP, SIOPErrors } from '../types';
+import { DiscoveryMetadataPayload, RPRegistrationMetadataPayload, SubjectIdentifierType } from '../types/SIOP.types';
+
+export function assertValidMetadata(opMetadata: DiscoveryMetadataPayload, rpMetadata: RPRegistrationMetadataPayload) {
```

Comment:
Right now we are doing verification, which obviously is nice. What I would also like to know as a user is which did methods and credentials formats etc I can actually use (so the actual intersection result). Probably makes sense to return that is a new method which returns a new interface with the supported metadata between RP and OP

Reply:
Fixed!

Reply:
Will resolve this comment as I made remarks about the new methods